fedimintd/src/lib.rs

```diff
@@ -6,16 +6,358 @@
+    /// and direct, and the port should be open it in the firewall.
+    #[arg(long, env = FM_BIND_P2P_ENV, default_value = "0.0.0.0:8173")]
+    bind_p2p: SocketAddr,
+
+    /// Address we bind to for the API
+    ///
+    /// Should be `0.0.0.0:8174` most of the time, as api connectivity is public
+    /// and direct, and the port should be open it in the firewall.
+    #[arg(long, env = FM_BIND_API_ENV, default_value = "0.0.0.0:8174")]
+    bind_api: SocketAddr,
+
+    /// Address we bind to for exposing the Web UI
+    ///
+    /// Built-in web UI is exposed as an HTTP port, and typically should
+    /// have TLS terminated by Nginx/Traefik/etc. and forwarded to the locally
+    /// bind port.
+    #[arg(long, env = FM_BIND_UI_ENV, default_value = "127.0.0.1:8175")]
+    bind_ui: SocketAddr,
+
+    /// Our external address for communicating with our peers
+    ///
+    /// `fedimint://<fqdn>:8173` for TCP/TLS p2p connectivity (legacy/standard).
+    ///
+    /// Ignored when Iroh stack is used. (newer/experimental)
+    #[arg(long, env = FM_P2P_URL_ENV)]
+    p2p_url: Option<SafeUrl>,
+
+    /// Our API address for clients to connect to us
+    ///
+    /// Typically `wss://<fqdn>/ws/` for TCP/TLS connectivity (legacy/standard)
+    ///
+    /// Ignored when Iroh stack is used. (newer/experimental)
+    #[arg(long, env = FM_API_URL_ENV)]
+    api_url: Option<SafeUrl>,
+
+    #[arg(long, env = FM_ENABLE_IROH_ENV)]
+    enable_iroh: bool,
+
+    /// Enable tokio console logging
+    #[arg(long, env = FM_BIND_TOKIO_CONSOLE_ENV)]
+    bind_tokio_console: Option<SocketAddr>,
+
+    /// Enable jaeger for tokio console logging
+    #[arg(long, default_value = "false")]
+    with_jaeger: bool,
+
+    /// Enable prometheus metrics
+    #[arg(long, env = FM_BIND_METRCIS_ENV)]
+    bind_metrics: Option<SocketAddr>,
+
+    /// Number of checkpoints from the current session to retain on disk
+    #[arg(long, env = FM_DB_CHECKPOINT_RETENTION_ENV, default_value = "1")]
+    db_checkpoint_retention: u64,
+}
+
+/// Block the thread and run a Fedimintd server
+///
+/// # Arguments
+///
+/// * `modules_fn` - A function to initialize the modules.
+///
+/// * `code_version_hash` - The git hash of the code that the `fedimintd` binary
+///   is being built from. This is used mostly for information purposes
+///   (`fedimintd version-hash`). See `fedimint-build` crate for easy way to
+///   obtain it.
+///
+/// * `code_version_vendor_suffix` - An optional suffix that will be appended to
+///   the internal fedimint release version, to distinguish binaries built by
+///   different vendors, usually with a different set of modules. Currently DKG
+///   will enforce that the combined `code_version` is the same between all
+///   peers.
+pub async fn run(
+    modules_fn: fn(
+        Network,
+    ) -> (
+        ServerModuleInitRegistry,
+        ServerModuleConfigGenParamsRegistry,
+    ),
```

Comment:
Why can't it just be the two registries instead of a fn? Whoever is calling run can surely inject the network themselves if needed (and any additional parameters that might be required)? 

Instead, e.g. `default_modules` could take the network as an argument.